pkgs/development/compilers/abcl/default.nix

```diff
@@ -36,31 +37,29 @@ stdenv.mkDerivation (finalAttrs: {
   buildPhase = ''
     runHook preBuild
 
-    ant
+    ant \
+      -Dabcl.runtime.jar.path="$out/lib/abcl/abcl.jar" \
+      -Dadditional.jars="$out/lib/abcl/abcl-contrib.jar"
 
     runHook postBuild
   '';
 
   installPhase = ''
     runHook preInstall
 
-    mkdir -p "$out"/{share/doc/abcl,lib/abcl}
+    mkdir -p "$out"/{share/doc/abcl,lib/abcl,bin}
     cp -r README COPYING CHANGES examples/  "$out/share/doc/abcl/"
     cp -r dist/*.jar contrib/ "$out/lib/abcl/"
-
-    makeWrapper ${jre}/bin/java $out/bin/abcl \
-      --add-flags "-classpath $out/lib/abcl/\*" \
-      ${lib.optionalString (lib.versionAtLeast jre.version "17")
-        # Fix for https://github.com/armedbear/abcl/issues/484
-        "--add-flags --add-opens=java.base/java.util.jar=ALL-UNNAMED \\"
-      }
-      --add-flags org.armedbear.lisp.Main
+    cp abcl $out/bin
 
     runHook postInstall
   '';
 
   passthru.updateScript = ./update.sh;
 
+  # Make sure cffi loads. It requires JNA.
+  passthru.tests.cffi = pkgs.abcl.pkgs.cffi;
```

Comment:
hmm yup - we need to add pkgs, withPackages in passthru instead of with wrapLisp in all-packages.nix. Previously I didn't know how to do that, but now with finalAttrs, it should work. However it can be another PR